app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt

```diff
@@ -142,6 +146,49 @@ class StoryFragmentPresenter @Inject constructor(
             ).parseOppiaHtml(
               storyItemViewModel.summary, binding.chapterSummary
             )
+
+          val str = if (storyItemViewModel.chapterSummary.chapterPlayState ==
+            ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES
+          ) {
+            String.format(
+              HtmlCompat.toHtml(
+                SpannedString
+                  .valueOf(view.context.getText(R.string.unlock_chapter)),
+                HtmlCompat.TO_HTML_PARAGRAPH_LINES_CONSECUTIVE
+              ),
+              storyItemViewModel.index,
+              storyItemViewModel.missingPrerequisiteChapter.name
+            )
+          } else {
+            String.format(
+              HtmlCompat.toHtml(
+                SpannedString.valueOf(
+                  view.context
+                    .getText(R.string.chapter_name)
+                ),
+                HtmlCompat.TO_HTML_PARAGRAPH_LINES_CONSECUTIVE
+              ),
+              storyItemViewModel.index + 1,
+              storyItemViewModel.name
+            )
+          }
+          binding.storyChapterCard.strokeWidth =
+            when (storyItemViewModel.chapterSummary.chapterPlayState) {
+              ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES ->
+                view.context.resources.getDimension(R.dimen.space_0dp).toInt()
+              else ->
+                view.context.resources.getDimension(R.dimen.story_chapter_stroke_width).toInt()
+            }
+          binding.titleContent =
+            htmlParserFactory.create(
+              resourceBucketName,
+              entityType,
+              storyItemViewModel.storyId,
+              imageCenterAlign = true
+            ).parseOppiaHtml(
+              str,
+              binding.chapterTitle
+            )
```

Comment:
I believe this can be done with Data binding only, we want to abstract away this as much as possible, so try doing it with data binding only.

Reply:
Sure.